misk/src/main/kotlin/misk/cloud/aws/logging/CloudwatchTrail.kt

```diff
@@ -0,0 +1,78 @@
+package misk.cloud.aws.logging
+
+import ch.qos.logback.classic.Logger
+import ch.qos.logback.classic.filter.ThresholdFilter
+import com.amazonaws.auth.AWSStaticCredentialsProvider
+import com.amazonaws.auth.AnonymousAWSCredentials
+import com.amazonaws.client.builder.AwsClientBuilder.EndpointConfiguration
+import com.amazonaws.services.logs.AWSLogs
+import com.amazonaws.services.logs.AWSLogsClientBuilder
+import com.google.common.annotations.VisibleForTesting
+import com.squareup.moshi.Moshi
+import misk.environment.InstanceMetadata
+import org.slf4j.LoggerFactory
+import java.time.Clock
+import java.util.concurrent.ArrayBlockingQueue
+import java.util.concurrent.BlockingQueue
+
+/** Enables logging for Cloudwatch Trail */
+object CloudwatchTrail {
+  /** Enable Cloudwatch Trail logging for the given app */
+  fun enableLogging(
```

Comment:
where does this get enabled?

Reply:
Right now it has to be explicitly called from the top level application code of services running in AWS. Ultimately though I want to have something like an ``installCloudSupport`` method that figures out what cloud you are on and installs all of the support infra to tie into that cloud's environmental systems.